NetKVM/ProtocolService/ProtocolService.cpp

```diff
@@ -627,10 +627,73 @@ class CVirtioAdaptersArray : public CAtlArray<CVirtioAdapter>
     }
 };
 
-static bool UninstallProtocol()
+class CFileFinder
+{
+public:
+    CFileFinder(const CString& WildCard) : m_WildCard(WildCard) {};
+    template<typename T> bool Process(T Functor)
+    {
+        HANDLE h = FindFirstFile(m_WildCard, &m_fd);
+        if (h == INVALID_HANDLE_VALUE)
+        {
+            return false;
+        }
+        while (Functor(m_fd.cFileName) && FindNextFile(h, &m_fd)) {}
+        FindClose(h);
+        return true;
+    }
+private:
+    WIN32_FIND_DATA m_fd = {};
+    const CString& m_WildCard;
+};
+
+class CInfDirectory : public CString
+{
+public:
+    CInfDirectory()
+    {
+        WCHAR* p = new WCHAR[MAX_PATH];
+        if (p)
+        {
+            if (GetWindowsDirectory(p, MAX_PATH))
+            {
+                Append(p);
+                Append(L"\\INF\\");
```

Comment:
Check how this will work with moving to isolation driver requirement (having DirID %13% in INF for driver installation).

Reply:
This is just INF uninstall, this will work the same even with with driver isolation